src/server.js

```diff
@@ -1,48 +1,133 @@
+    socket.emit(events.SERVER_ADD_PLAYERS, await player.getAll())
 
-  io.on(CONNECTION, async (socket) => {
-    const newPlayer = {
-      name: NAMES[rand(0, NAMES.length - 1)],
-      x: randPos(1, 7),
-      y: randPos(1, 6),
-    }
+    // When new player is connected
+    socket.on(events.CLIENT_PLAYER_CONNECT, async ({ name }) => {
+      logger.info(`Player connected ''${name}'': ${socket.id}`)
 
-    player.insert(newPlayer)
-    addBodies(createBox(newPlayer.x, newPlayer.y))
-    start = true
+      // Store new player and with its socket ID
+      const newPlayer = {
+        id: socket.id,
+        name,
+        color: utils.randomColor(),
+        x: utils.random(1, 7),
+        y: utils.random(1, 6),
+        latency: 0,
+      }
+      await player.insert(newPlayer)
+      socket.emit(events.SERVER_SET_PLAYER, socket.id)
 
-    // Connect events
-    socket.emit(SET_PLAYERS, await player.getAll())
-    player.onChange(p => io.emit(UPDATE_PLAYER, p))
-    socket.on(UPDATE_PLAYER, player.update) // Should be checked (for cheating player)
+      // Send players to all clients
+      io.emit(events.SERVER_ADD_PLAYERS, await player.getAll())
+    })
+
+    // When the game is started
+    socket.on(events.CLIENT_START_GAME, async () => {
+      // Re-init values if restarted
+      players = []
+      if (physicInterval) clearInterval(physicInterval)
+      if (synchroInterval) clearInterval(synchroInterval)
+
+      // Initialize physic server
+      physic.init()
+      physicInterval = setInterval(physic.tick, (1 / 60) * 1000)
+
+      // Add players to physic engine
+      const playersConnected = await player.getAll()
+      playersConnected.forEach(p => players.push({ ...p, body: physic.createBox(p.x, p.y) }))
+
+      // Set server state to 'STARTED'
+      setGameState(GAME_STARTED)
+
+      // Synchronize clients with server physics
+      synchroInterval = setInterval(async () => {
+        io.emit(events.SERVER_SYNCHRONIZE, players.map(p => ({
+          id: p.id,
+          force: p.body.force,
+          position: p.body.position,
+          velocity: p.body.velocity,
+          angle: p.body.angle,
+          angularForce: p.body.angularForce,
+          angularVelocity: p.body.angularVelocity,
+        })))
+      }, 200)
+    })
+
+    // When client moves a player
+    socket.on(events.CLIENT_MOVE, ({ id, velocity }) => {
+      const { body } = find(players, { id })
+      p2.vec2.add(body.velocity, body.velocity, velocity)
+    })
+
+    // When a client is disconnected
+    socket.on(events.DISCONNECT, async () => {
+      logger.info(`Client disconnected : ${socket.id}`)
+      // delete player from DB
+      await player.delete(socket.id)
+      // check if enought players to play
+      const connectedPlayers = await player.getAll()
+      if (connectedPlayers.length <= 1) setGameState(GAME_WAITING)
+      // Send players to all clients
+      io.emit(events.SERVER_ADD_PLAYERS, connectedPlayers)
```

Comment:
As it is implemented in `dbAbstract.js`, `player.onChange` returns `null` when a deletion is triggered

We may should implement `onInsert`, `onDelete`, `onModify` instead of `onChange` in `dbAbstract.js`